api/clients/codecs/default_blob_codec.go

```diff
@@ -0,0 +1,55 @@
+package codecs
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+
+	"github.com/Layr-Labs/eigenda/encoding/utils/codec"
+)
+
+type DefaultBlobCodec struct{}
+
+var _ BlobCodec = DefaultBlobCodec{}
+
+func NewDefaultBlobCodec() DefaultBlobCodec {
+	return DefaultBlobCodec{}
+}
+
+func (v DefaultBlobCodec) EncodeBlob(rawData []byte) ([]byte, error) {
+	codecBlobHeader := make([]byte, 32)
+	// first byte is always 0 to ensure the codecBlobHeader is a valid bn254 element
+	// encode version byte
+	codecBlobHeader[1] = byte(DefaultBlobEncoding)
+
+	// encode length as uint32
+	binary.BigEndian.PutUint32(codecBlobHeader[2:6], uint32(len(rawData))) // uint32 should be more than enough to store the length (approx 4gb)
+
+	// encode raw data modulo bn254
+	rawDataPadded := codec.ConvertByPaddingEmptyByte(rawData)
+
+	// append raw data
+	encodedData := append(codecBlobHeader, rawDataPadded...)
+
+	return encodedData, nil
+}
+
+func (v DefaultBlobCodec) DecodeBlob(data []byte) ([]byte, error) {
```

Comment:
Any reason why we aren't doing a length invariant on the data input? If not mistaken this could index out of bounds